### DISCUSSION TITLE
Custom subclass | Missing property on cloning *DONE*

### DISCUSSION THREAD

**Discussion by timoostrich** [fabricjs/fabric.js#7038] [Q&A]
**EDIT: Found my mistake. The property needs to be an object but I added an array...** 

I really need some help with a custom class in my project.

I created a custom subclass, which bounces and rotates letters by some random values. The text comes from a iText box and I am "converting" that box to my custom object (called bouncedText).

Everything works fine, but I am not able to make the cloning work.
The dev setup creates an itext box and converts it to bouncedText automatically.

In the console you can see a custom property of my bouncedText object. It's an array containing random numbers. I am using these numbers in my _render function to place the letters.

But when I want to clone that object (just **click on the lower middle icon**) you will get an error.
The property "randNumbers" is now undefined. Although copying my other custom properties (like bounceIntensity and rotationIntensity) works.

What am I doing wrong!?

I also tried to do the random generation stuff in the object itself. That worked - but on clone the object gets new random numbers. That should not happen. I also need to be able to save and load (JSON) without changed random numbers.

I'm stuck here - please help. :-) Thanks in advance!

**Comment by asturur** [comment]
There is no link to any example. Anyway you have likely to call JSON.stringify on your array before serializing it.

**Reply by timoostrich** [reply]
Just edited my post this morning. I added a custom object property as an array. That worked till I needed to clone. Using an object works perfect. Thanks!

**Reply by asturur** [reply]
i m not sure object or array males difference

**Reply by timoostrich** [reply]
In my case it did, I did not changing anything else.
Or maybe my clone function isn't/wasn't right?
`		fabric.BouncedText.fromObject = function(object, callback) {
			return fabric.Object._fromObject('BouncedText', object, callback, 'BouncedText');
		}`
